app/controllers/ems_common.rb

```diff
@@ -648,13 +647,17 @@ def form_instance_vars
     @emstype_display = model.supported_types_and_descriptions_hash[@ems.emstype]
   end
 
-  def get_regions(provider_name)
-    regions = {}
-    "ManageIQ::Providers::#{provider_name}::Regions".constantize.all.each do |region|
-      regions[region[:name]] = region[:description]
+  def retrieve_provider_regions
+    cloud_managers = model.supported_subclasses.select { |c| c.is_available?(:regions) }
+    cloud_managers.each_with_object({}) do |cloud_manager, provider_regions|
+      regions = cloud_manager.parent::Regions.all.sort_by { |r| r[:description] }
```

Comment:
Is there no dot access here on the region objects (I can't remember)?  If not, we should put it on our TODO list to do that, so that one could substitute out objects (particular since it *looks* like we're querying db resources).